### PR TITLE
MySQL: Reworked Varchar/Varbinary column identification

### PR DESCRIPTION
**Pull request by Rentacookie**
MySQL uses the same type code for Varbinary and Varchar columns. This then requires additional interrogation of the column metadata to correctly determine the type. Previously we just checked the binary flag in the column metadata to determine if a column was in fact Varbinary. But it turns out Varchar columns with binary encoding set will also have the binary flag set, leading to mis-identifcation. Instead we now check the characterSet for the column for the binary characterSet which seems to only be set for actual Varbinary columns

### CODE REVIEW

File: modules/module-mysql/test/src/mysql-to-sqlite.test.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ describe('MySQL Data Types', () => {
 
     char_col CHAR(10),
     varchar_col VARCHAR(255),
+    varchar_binary_encoding_col VARCHAR(255) COLLATE utf8mb4_bin,
```
Comment (Rentacookie) [inline modules/module-mysql/test/src/mysql-to-sqlite.test.ts:43]: Specifying COLLATE for a column without specifying the character set defaults the column characterset to the one for COLLATE:
https://dev.mysql.com/doc/refman/5.7/en/charset-column.html

Comment (rkistner) [inline modules/module-mysql/test/src/mysql-to-sqlite.test.ts:43]: `binary_encoding` doesn't sound quite correct here. As I understand it, this is utf8 encoding, but comparison is byte-wise.

Comment (rkistner) [reply modules/module-mysql/test/src/mysql-to-sqlite.test.ts:43]: Could we also add a test for something like `VARCHAR(255) CHARACTER SET binary`?

Comment (Rentacookie) [reply modules/module-mysql/test/src/mysql-to-sqlite.test.ts:43]: Thanks - the edge case `VARCHAR(255) CHARACTER SET binary` highlighted a problem that there it not enough info in the metadata to correctly deduce the type. 
Looks like the only way is to query the table schema directly.

Comment (Rentacookie) [reply modules/module-mysql/test/src/mysql-to-sqlite.test.ts]: Turns out `VARCHAR(255) CHARACTER SET binary` is essentially just a `Varbinary` column, therefore we parse it as one.

**Review by rkistner — APPROVED** [review]
